src/components/ChatLog.jsx

```diff
@@ -0,0 +1,53 @@
+import './ChatLog.css';
+import PropTypes from 'prop-types';
+import ChatEntry from './ChatEntry';
+
+const ChatLog = ({
+  entries,
+  onLikeChange = () => { },
+  userColors = { local: 'black', remote: 'black' },
+  localUser = '',
+  likedMessages = []
+}) => (
+  <ul className="chat-log">
+    {entries.map(({ id, sender, body, timeStamp, liked }) => {
+      const isLocal = sender === localUser;
+      const textColor = isLocal ? userColors.local : userColors.remote;
+
+      return (
+        <ChatEntry
+          key={id}
+          id={id}
+          sender={sender}
+          body={body}
+          timeStamp={timeStamp}
+          liked={likedMessages.includes(id) ? liked : false}
+          onLikeChange={onLikeChange}
+          isLocal={isLocal}
+          style={{ color: textColor }}
+        />
+      );
+    })}
+  </ul>
```

Comment:
More typically in industry you'll see the result of using map referenced by a variable and using that variable in the JSX, like:

```js
  const chatEntries = entries.map(); // rest of your logic for mapping over each entry.

  <ul className="chat-log">
    {chatEntries}
  </ul>